xbmc/addons/kodi-addon-dev-kit/tools/debian-addon-package-test.sh

```diff
@@ -0,0 +1,141 @@
+
+--------------------------------------------------------------------------------"
+}
+
+function checkEnv {
+  echo "#------ build environment ------#"
+  echo "BASE_DIR: $BASE_DIR"
+  echo "REPO_DIR: $REPO_DIR"
+  getVersion
+  echo "VERSION_MAIN: $VERSION_MAIN"
+  echo "VERSION_MINOR: $VERSION_MINOR"
+  echo "VERSION_REVISON: $VERSION_REVISON"
+  if [ $GIT_REV ]; then
+    echo "GIT_REV: $GIT_REV"
+  fi
+  echo "TAGREV: $TAGREV"
+  echo "DIST: $DIST"
+  echo "ARCHS: $ARCHS"
+
+  echo "#-------------------------------#"
+}
+
+function getVersion {
+  if [ -d ${BASE_DIR}/.git ]; then
+    getGitRev
+  else
+    REV="master"
+  fi
+  PACKAGEVERSION=$(cat ${BASE_DIR}/$REPO_DIR/addon.xml.in | sed -n '/version/ s/.*version=\"\([0-9]\+\.[0-9]\+\.[0-9]\+\)\".*/\1/p')
+  VERSION_MAIN=$(echo $PACKAGEVERSION | awk -F. '{print $1}')
+  VERSION_MINOR=$(echo $PACKAGEVERSION | awk -F. '{print $2}')
+  VERSION_REVISON=$(echo $PACKAGEVERSION | awk -F. '{print $3}')
+}
+
+function getGitRev {
+    cd $BASE_DIR || exit 1
+    GIT_REV=$(git log -1 --pretty=format:"%h")
+}
+
+function cleanup() {
+  echo "Starting to remove debian generated files"
+  rm -rf ${BASE_DIR}/obj-$ARCHS-linux-gnu
+  rm -rf ${BASE_DIR}/debian/.debhelper
+  rm -rf ${BASE_DIR}/debian/"kodi-"$(echo $REPO_DIR | tr . -)
+  rm -rf ${BASE_DIR}/debian/"kodi-"$(echo $REPO_DIR | tr . -)-dbg
+  rm -rf ${BASE_DIR}/debian/tmp
+  rm -f ${BASE_DIR}/debian/changelog
+  rm -f ${BASE_DIR}/debian/debhelper-build-stamp
+  rm -f ${BASE_DIR}/debian/files
+  rm -f ${BASE_DIR}/debian/*.log
+  rm -f ${BASE_DIR}/debian/*.substvars
+}
+
+if [[ $1 = "-h" ]] || [[ $1 = "--help" ]]; then
+  echo "$0:"
+  usage
+  exit
+fi
+
+if [ ! $1 ] || [ ${1} = "--clean" ]; then
+  printf "$0:\nERROR: Addon source code must be given as the first parameter!\n"
+  usage
+  exit 1
+elif [ ! -d $1 ]; then
+  printf "$0:\nERROR: Given folder is not present or not a valid addon source!\n"
+  usage
+  exit 1
+fi
+
+ARCHS=$(uname -m)
+BASE_DIR=$(realpath ${1})
+REPO_DIR=$(basename $1)
+DIST=$(lsb_release -cs)
+
+if [ ! -f ${BASE_DIR}/$REPO_DIR/addon.xml.in ]; then
+  echo "$0:
+ERROR: \"required $REPO_DIR/addon.xml.in\" not found!
+
+The base source dir and addon.xml.in containing directory names must be equal"
+  usage
+  exit 1
+fi
+
+checkEnv
+
+ORIGTARBALL="kodi-"$(echo $REPO_DIR | tr . -)"_${PACKAGEVERSION}.orig.tar.gz"
+
+if [[ ${2} = "--clean" ]]; then
+  cleanup
+  exit
+fi
+
+echo "Detected PackageVersion: ${PACKAGEVERSION}"
+
+sed -e "s/#PACKAGEVERSION#/${PACKAGEVERSION}/g" \
+    -e "s/#TAGREV#/${TAGREV}/g" \
+    -e "s/#DIST#/${DIST}/g" ${BASE_DIR}/debian/changelog.in > ${BASE_DIR}/debian/changelog
+
+echo "Create needed compressed source code file: ${BASE_DIR}/../${ORIGTARBALL}.tar.gz"
+
+# git archive --format=tar.gz -o ${BASE_DIR}/../${ORIGTARBALL} HEAD # Unused git, leaved as optional way
+tar -zcvf ${BASE_DIR}/../${ORIGTARBALL} * # Used to prevent on code changed for test a git commit
```

Comment:
--exclude=.git/*

Reply:
Thanks! Have extended it a bit and .gitignore becomes also now scanned to prevent not wanted parts.